### DISCUSSION TITLE
[Valhalla] Issue with installation on MacOS

### DISCUSSION THREAD

**Discussion by veenits123** [valhalla/valhalla#3952] [General]
I am following this to install: https://valhalla.github.io/valhalla/building/#building-from-source-macos
I have followed all the steps above this, and everything worked, but I am unable to do this step: Then, build [prime_server](https://github.com/kevinkreiser/prime_server#build-and-install).

Here's the log:
```
vee_nits123@Veenits-MacBook-Air valhalla % git submodule update --init --recursive
vee_nits123@Veenits-MacBook-Air valhalla % ./autogen.sh
zsh: no such file or directory: ./autogen.sh
vee_nits123@Veenits-MacBook-Air valhalla % 
```
Any help would be appreciated.

@nilsnolde

**Comment by nilsnolde** [comment]
Well, https://github.com/kevinkreiser/prime_server is an entirely different project which needs to be built separately.. The README isn't mentioning it explicitly, but you will have to first `git pull https://github.com/kevinkreiser/prime_server` before attempting to build it.

**Reply by veenits123** [reply]
@nilsnolde can you please review once, that all my builds and installations are correct? Thanks!

Logs are here: https://github.com/veenits123/valhalla/issues/1

**Comment by hitenvidhani** [comment]
I am building valhalla on Linux and facing a similar issue. Could someone help me out?
![image](https://user-images.githubusercontent.com/60215053/217922660-213f809d-58c1-4bd5-bf1b-cec04d0ea439.png)

**Reply by nilsnolde** [reply]
Are you sure you followed the steps outlined in the README of https://github.com/kevinkreiser/prime_server? 

Please no screenshots of text..

**Reply by hitenvidhani** [reply]
I followed the steps in the README of prime_Server and it worked, I was just confused whether libprime and prime_server were the same packages.
Could you please tell how do I check if it's built successfully after following all the steps here (https://valhalla.github.io/valhalla/building/#building-from-source-linux) ?

**Reply by nilsnolde** [reply]
You run one of the executables or the tests.

**Reply by hitenvidhani** [reply]
I ran this command in the build directory for running all tests : `make check -j$(nproc)`
The logs show that some tests failed, what would be the reason?
Here is the log : 
`[  PASSED  ] 2 tests.
[  FAILED  ] 3 tests, listed below:
[  FAILED  ] UtilOdin.test_time
[  FAILED  ] UtilOdin.test_date
[  FAILED  ] UtilOdin.test_supported_locales
`

**Reply by hitenvidhani** [reply]
@nilsnolde I was setting up the server using the steps here : https://valhalla.github.io/valhalla/building/#running-valhalla-server-on-unix
These are the logs :
`hiten@hiten:~/Desktop/valhalla$ valhalla_service valhalla.json 1
valhalla_service: error while loading shared libraries: libprime_server.so.0: cannot open shared object file: No such file or directory `
Where should I look for the valhalla.json file or could you please tell how do I fix this?

**Reply by nilsnolde** [reply]
where did you install prime_server to? that path needs to be seen by your link loader..
